pulsar-client/src/main/java/org/apache/pulsar/client/impl/Backoff.java

```diff
@@ -40,19 +42,33 @@
 
     @VisibleForTesting
     Backoff(long initial, TimeUnit unitInitial, long max, TimeUnit unitMax, long mandatoryStop,
-            TimeUnit unitMandatoryStop, Clock clock) {
+            TimeUnit unitMandatoryStop, Clock clock, long backoffIntervalNanos, long maxBackoffIntervalNanos) {
```

Comment:
since we are introducing new fields in the constructor and getting more and more constructors, it might be a good time to rewrite this class to use Builder pattern.